README.md

```diff
@@ -768,14 +797,22 @@ This is useful for when you need to do a subsequent computation using the inner
 **Signature:**
 
 ```typescript
-type AndThenFunc = (t:  T) => ResultAsync<U, E> | Result<U, E>
-andThen<U>(f: AndThenFunc): ResultAsync<U, E> { ... }
+class ResultAsync<T, E> {
+  andThen<U>(
+    callback: (value: T) => Result<U, E> | ResultAsync<U, E>
+  ): ResultAsync<U, E> { ... }
```

Comment:
```suggestion
  andThen<U, F>(
    callback: (value: T) => Result<U, F> | ResultAsync<U, F>
  ): ResultAsync<U, E | F> { ... }
```

Reply:
This is exactly what the second signature is.